### ISSUE TITLE
Compile-AppInNavContainer() compatibility fix for alc.exe interaction

### ISSUE THREAD

**Issue by fabat** [microsoft/navcontainerhelper#1150]
**Compile-AppInNavContainer():379** was recently improved by enclosing paths provided to alc.exe in double-quotes.
> $alcParameters = @("/project:""$appProjectFolder""", "/packagecachepath:""$appSymbolsFolder""", "/out:""$appOutputFile""")

This can cause alc.exe to fail execution for paths ending in a single DirectorySeparatorChar . At least alc.exe argument **/project** has proven to be problematic on my site: https://github.com/microsoft/AL/issues/4456

Compile-AppInNavContainer() could implicitly prevent such a situation by normalizing provided paths to either end without  or with the required pair of DirectorySeparatorChars.

**Potential Fix**
I've chosen to replace a single, terminal DirectorySeparatorChar with a pair accordingly, but removing it seems to be fine as well.
```
function normalize()
{
    # https://github.com/microsoft/AL/issues/4456
    [CmdletBinding()]
    Param(
        [Parameter(Mandatory=$true)]
        [ValidateNotNullOrEmpty()]
        [string] $path
    )
    $normalizedPath = (Convert-Path $path)
    return $(if($normalizedPath.EndsWith([System.IO.Path]::DirectorySeparatorChar)) {
        "$normalizedPath$([System.IO.Path]::DirectorySeparatorChar)"
    }else{
        $normalizedPath
    })
}

# Prepare test-data

if($env:OS.ToLower().contains('windows')) {    
    $PathA = 'C:/tmp/fabian/projectdir'
    $PathB = 'C:/tmp/fabian/projectdir/'
    $PathC = 'C:/tmp/fabian/projectdir//'
    $PathD = 'C:\tmp\fabian\projectdir'
    $PathE = 'C:\tmp\fabian\projectdir\'
    $PathF = 'C:\tmp\fabian\projectdir\\'
} else {
    # assuming MacOS
    $PathA = '/Users/fabian/projectdir'
    $PathB = '/Users/fabian/projectdir/'
    $PathC = '/Users/fabian/projectdir//'
    $PathD = '\Users\fabian\projectdir'
    $PathE = '\Users\fabian\projectdir\'
    $PathF = '\Users\fabian\projectdir\\'
}

$PathCollection = @($PathA, $PathB, $PathC, $PathD, $PathE, $PathF)

# Properly format paths to meet requirements as per https://github.com/microsoft/AL/issues/4456
foreach ($path in $PathCollection) {
    Write-Host $(normalize($path))
}
```

**Output**
```
C:\tmp\fabian\projectdir
C:\tmp\fabian\projectdir\\
C:\tmp\fabian\projectdir\\
C:\tmp\fabian\projectdir
C:\tmp\fabian\projectdir\\
C:\tmp\fabian\projectdir\\
```
best
fabian

**Comment by freddydk** [comment]
Wouldn't it be easier just to use .TrimEnd('/\') on the paths to remove trailing slashes or backslashes?
Doubling the slashes seems to be a workaround to a problem, which eventually some day might work differently I think.

**Comment by fabat** [comment]
yes, it would. Depending on what happens afterwards with the path. In my case there is lot's of upcoming code where I can not be certain whether some portion wasn't relying on at least a trailing DirectorySeparatorChar at some point for branching. So I decided to go with a DirectorySeparatorChar pair, but as mentioned... alc.exe itself seems to be perfectly happy wither either:
- no DirectorySeparatorChar
- or a pair of them

**Comment by freddydk** [comment]
Obviously, the problem is that the \ in front of the " will escape the ". So the problem isn't really in the path.
I will proceed to trim end string to avoid the escape.

**Comment by freddydk** [comment]
Shipped in 0.7.0.19
